### PR TITLE
Fix `Update Spring Cloud Azure Support File` action failed

### PR DESCRIPTION
**Pull request by Netyyyy**
As title. 
Fix action [failed](https://github.com/Azure/spring-cloud-azure-tools/actions/runs/4716098675/jobs/8363509528#step:4:8924) when the new Spring Boot doesn't have a supported Spring Cloud version in [metadata](https://start.spring.io/actuator/info)
image

### CODE REVIEW

File: src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java [hunk 1]
```diff
@@ -115,13 +115,18 @@ private String findCompatibleSpringCloudVersion(String springBootVersion) {
...
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getValue().match(Version.parse(springBootVersion)))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList())
+                .stream().findFirst().orElseThrow();
+        } catch (Exception e) {
+            LOGGER.info("Spring Boot " + springBootVersion + " does not support yet, so skip it");
+        }
+        return "not get the supported Spring Cloud version yet";
```
Comment (saragluna) [inline src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java:129]: Could you explain this logic, why the error message is like the Spring Boot version is not supported yet?

Comment (Netyyyy) [reply src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java:129]: We get a compatible Spring Cloud version from [metadata](https://start.spring.io/actuator/info) for each Spring Boot version, But when a new Spring Boot version was published [here](https://api.spring.io/project_metadata/spring-boot), the supported version may not update yet, for example, Spring Boot version `3.1.0-M2` and `3.1.0-SNAPSHOT` not supported in metadata now:
image

File: src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java [hunk 2]
```diff
@@ -115,13 +115,18 @@ private String findCompatibleSpringCloudVersion(String springBootVersion) {
...
-            .collect(Collectors.toList())
-            .stream().findFirst().orElseThrow();
+        try {
+            return this.springCloudCompatibleSpringBootVersionRanges
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getValue().match(Version.parse(springBootVersion)))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList())
+                .stream().findFirst().orElseThrow();
```
Comment (saragluna) [inline src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java:125]: If you are catching the exception here, how about not throwing it in the first place? And what's the caller of the function going to do with the string "not get the supported Spring Cloud version yet" you return?

File: src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java [hunk 3]
```diff
@@ -115,13 +115,18 @@ private String findCompatibleSpringCloudVersion(String springBootVersion) {
...
-            .collect(Collectors.toList())
-            .stream().findFirst().orElseThrow();
+        try {
+            return this.springCloudCompatibleSpringBootVersionRanges
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getValue().match(Version.parse(springBootVersion)))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList())
+                .stream().findFirst().get();
```
Comment (saragluna) [inline src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java:125]: Would it be better to return `orElse(xxx)`?

File: src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java [hunk 4]
```diff
@@ -115,13 +115,18 @@ private String findCompatibleSpringCloudVersion(String springBootVersion) {
...
+        } catch (Exception e) {
+            LOGGER.info("Spring Boot " + springBootVersion + " does not support yet, so skip it");
+        }
+        return "not get the supported Spring Cloud version yet";
```
Comment (saragluna) [inline src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java:129]: And how about return `NONE_SUPPORTED_SPRING_CLOUD_VERSION`?

**Review by saragluna — APPROVED** [review]

**Pull request merged**